arch/arc/core/thread.c

```diff
@@ -178,6 +178,7 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 #ifdef CONFIG_ARC_USE_UNALIGNED_MEM_ACCESS
 	iframe->status32 |= _ARC_V2_STATUS32_AD;
 #endif
+	iframe->status32 |= _ARC_V2_STATUS32_DZ;
```

Comment:
Why not add this to the initial assignments above, as there is no conditional involved.

Reply:
Like this ?

```
#ifdef CONFIG_USERSPACE
	/* enable US bit, US is read as zero in user mode. This will allow user
	 * mode sleep instructions, and it enables a form of denial-of-service
	 * attack by putting the processor in sleep mode, but since interrupt
	 * level/mask can't be set from user space that's not worse than
	 * executing a loop without yielding.
	 */
	iframe->status32 = _ARC_V2_STATUS32_US | _ARC_V2_STATUS32_DZ;
	if (is_user(thread)) {
		iframe->pc = (uint32_t)z_user_thread_entry_wrapper;
	} else {
		iframe->pc = (uint32_t)z_thread_entry_wrapper;
	}
#else
	iframe->status32 = _ARC_V2_STATUS32_DZ;
	iframe->pc = ((uint32_t)z_thread_entry_wrapper);
#endif /* CONFIG_USERSPACE */
```

Reply:
Yes, that's what I had in mind.